packages/drivers/odsp-driver/src/test/localOdspDriver.spec.ts

```diff
@@ -0,0 +1,240 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "fs";
+import { strict as assert } from "assert";
+import { IStream } from "@fluidframework/driver-definitions";
+import { IOdspResolvedUrl } from "@fluidframework/odsp-driver-definitions";
+import {
+    IClient,
+    ISequencedDocumentMessage,
+    MessageType,
+    SummaryType,
+} from "@fluidframework/protocol-definitions";
+import { MockLogger } from "@fluidframework/telemetry-utils";
+import { LocalOdspDocumentService } from "../odspDocumentService";
+import { LocalOdspDocumentServiceFactory } from "../odspDocumentServiceFactory";
+import { LocalOdspDocumentStorageService } from "../odspDocumentStorageManager";
+
+describe("Local Odsp driver", () => {
+    // TODO: add end-to-end test
+
+    const fakeOdspResolvedUrl: IOdspResolvedUrl = {
+        type: "fluid",
+        odspResolvedUrl: true,
+        id: "1",
+        siteUrl: "fakeUrl",
+        driveId: "1",
+        itemId: "1",
+        url: "fakeUrl",
+        hashedDocumentId: "1",
+        endpoints: {
+            snapshotStorageUrl: "fakeUrl",
+            attachmentPOSTStorageUrl: "fakeUrl",
+            attachmentGETStorageUrl: "fakeUrl",
+            deltaStorageUrl: "fakeUrl",
+        },
+        tokens: {},
+        fileName: "fakeName",
+        summarizer: false,
+        fileVersion: "1",
+    };
+
+    const localSnapshot = fs.readFileSync(
+        `${__dirname}/../../src/test/localSnapshots/localSnapshot1.json`,
+        { encoding: "utf8" },
+    );
+
+    async function assertThrowsAsync(fn: () => Promise<any>) {
+        try {
+            await fn();
+            assert.fail("Expected exception to be thrown");
+        } catch (err) { }
+    }
+
+    /**
+     * ! Will re-throw error when encountered
+     * ! This function helps explicitly define when we don't expect errors
+     */
+    async function assertDoesNotThrowAsync(fn: () => Promise<any>) {
+        await fn();
+    }
+
+    describe("Local Odsp document service factory", () => {
+        it("Can use a real snapshot", () => {
+            assert.doesNotThrow(() => {
```

Comment:
I feel like the only usefulness of assert.doesNotThrow() are
- providing custom message (if it throws).
- used in assert.throws() scope and ensure that particular statement does not throw.
Otherwise directly calling API has same effect, and is more readable, no?

Reply:
I like the usage of `assert.doesNotThrow` to be explicit and to indicate what the intention of a call is.

Reply:
Got it. I guess it's fine :)